### PR TITLE
Load / save example

### PR DESCRIPTION
**Pull request by gemeinl**
- added an example to showcase saving and loading of braindecode datasets
- minor improvements to serialization functions

### CODE REVIEW

File: examples/plot_load_save_datasets.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+"""Load and save dataset example
+================================
```
Comment (agramfort) [inline examples/plot_load_save_datasets.py:2]:
```suggestion
"""
Load and save dataset example
=======================
```

**Review by agramfort — COMMENTED** [review]
did you consider a different design where you could do:

ds.save(path='./')

load_dataset(path='./')

it would allow to save the object without a new function to import
and this API would be the same for all datasets (no need to say  "concat").

**Comment by gemeinl** [comment]
No, we have not thought about it yet, but I like it.
Would it be an overkill or confusing to offer both? Due to the loading function, we will not get rid of the serialization module, so we could just call `save_concat_dataset` from within `ds.save()`, right?

The thing with 'concat' is always a bit tricky to me. I named the functions like this to be precise about what's going on. But actually, whenever we talk about datasets in braindecode, we mean concat of datasets, either holding mne raws or mne eopchs. I am always very insecure how to deal with this. For me it would be fine to just always call it 'dataset' If I am not mistaken, the BaseDataset is never really exposed to users, so there should not be too much confusion. What do you think?

**Comment by agramfort** [comment]
>
> No, we have not thought about it yet, but I like it.
> Would it be an overkill or confusing to offer both? Due to the loading
> function, we will not get rid of the serialization module, so we could just
> call save_concat_dataset from within ds.save(), right?
>
$python -c "import this" | grep preferably
There should be one-- and preferably only one --obvious way to do it.

so I would favor only one way with the .save method

> The thing with 'concat' is always a bit tricky to me. I named the
> functions like this to be precise about what's going on. But actually,
> whenever we talk about datasets in braindecode, we mean concat of datasets,
> either holding mne raws or mne eopchs. I am always very insecure how to
> deal with this. For me it would be fine to just always call it 'dataset' If
> I am not mistaken, the BaseDataset is never really exposed to users, so
> there should not be too much confusion. What do you think?
>

agreed ! we always concatenate sessions or subjects etc.

maybe others have some opinion too...

**Comment by robintibor** [comment]
Great, nice to have a documented example. During sprint we should consider how to make overall doc/tutorials nicely structured so peoplesee things easily.

**Pull request merged**